catalog/dags/es/create_new_es_index/create_new_es_index.py

```diff
@@ -0,0 +1,172 @@
+    # we do not necessarily know the index_name so we cannot access the configuration
+    # directly by key. We instead get the first value from the dict, knowing that we
+    # have already ensured in a previous check that there is exactly one value in the
+    # response.
+    config = next(iter(response.values()))
+    return config
+
+
+@task
+def merge_index_configurations(new_index_config, current_index_config):
+    """
+    Merge the `new_index_config` into the `current_index_config`, and
+    return an index configuration in the appropriate format for being
+    passed to the `create_index` API.
+    """
+    # Do not automatically apply any aliases to the new index
+    current_index_config.pop("aliases")
+
+    # Remove fields from the current_index_config that should not be copied
+    # over into the new index (such as uuid)
+    for setting in EXCLUDED_INDEX_SETTINGS:
+        current_index_config.get("settings", {}).get("index", {}).pop(setting)
+
+    # Merge the new configuration values into the current configuration
+    return merge_configurations(current_index_config, new_index_config)
+
+
+@task
+def get_final_index_configuration(
+    override_config: bool,
+    # The new config which was passed in via DAG params
+    index_config,
+    # The result of merging the index_config with the current index config.
+    # This may be None if the merge tasks were skipped using the override param.
+    merged_config,
+    index_name: str,
+):
+    """
+    Resolve the final index configuration to be used in the `create_index`
+    task.
+    """
+    config = index_config if override_config else merged_config
+
+    # Apply the desired index name
+    config["index"] = index_name
+    return config
+
+
+@task
+def create_index(index_config, es_host: str):
+    hook = ElasticsearchPythonHook(hosts=[es_host])
+    es_conn = hook.get_conn
+
+    new_index = es_conn.indices.create(**index_config)
+
+    return new_index
+
+
+@task_group(group_id="trigger_and_wait_for_reindex")
+def trigger_and_wait_for_reindex(
+    index_name: str, source_index: str, query: dict, timeout: timedelta, es_host: str
+):
+    @task
+    def trigger_reindex(index_name: str, source_index: str, query: dict, es_host: str):
+        hook = ElasticsearchPythonHook(hosts=[es_host])
+        es_conn = hook.get_conn
+
+        source = {"index": source_index}
+        # An empty query is not accepted; only pass it
+        # if a query was actually supplied
+        if query:
+            source["query"] = query
+
+        response = es_conn.reindex(
+            source=source,
+            dest={"index": index_name},
+            # Parallelize indexing
+            slices="auto",
+            # Do not hold the slot while awaiting completion
+            wait_for_completion=False,
+            # Immediately refresh the index after completion to make
+            # the data available for search
+            refresh=True,
```

Comment:
Sure, great idea! It's a bit weird at the moment because DAGs like the data refresh that perform reindexing on the ingestion server will have their throttling controlled by a different environment variable on the ingestion server itself

I thought about modifying the way the data refresh does it to send this variable to the ingestion server, just so that everything is consistent. But if the data refresh gets moved into Airflow, this will be a non-issue so I think it's fine for now :)

Updated in [cabffa4](https://github.com/WordPress/openverse/pull/3537/commits/cabffa46e9ac8cc92b2d852c33c46413276ee18e) to read from the airflow variable, but it can be overridden per-environment if we decide that's necessary.